generic.go

```diff
@@ -112,15 +103,18 @@ func NewGeneric(config *Config, queue workqueue.RateLimitingInterface, workers i
 
 			controllers[descr.Gvk] = constructed.Interface
 
-			objectProcessTime := prometheus.NewHistogram(
+			// objectProcessTime := prometheus.NewHistogram(
+			objectProcessTime := prometheus.NewHistogramVec(
 				prometheus.HistogramOpts{
-					Namespace: constructorConfig.AppName,
-					Name:      fmt.Sprintf("process_%s_seconds", objectName),
-					Help:      fmt.Sprintf("Histogram measuring the time it took to process a %s", &groupKind),
+					Namespace: metricsNamespace,
+					Name:      "process_object_seconds",
+					Help:      "Histogram measuring the time it took to process an object",
 				},
+				[]string{"controller", "object"},
```

Comment:
Object name without namespace of that object is not as useful as it could be. Add namespace?